### PR TITLE
rudimentary profile drawing, need some updates on math

### PR DESCRIPTION
**Pull request by tcasewiley**

### CODE REVIEW

File: src/my-element.ts [hunk 1]
```diff
@@ -115,6 +115,8 @@ export class MyElement extends LitElement {
   @state()
   save = false
 
+  
+  private cachedImages: { [key: number]: HTMLImageElement } = {}; //Here we define the structure for the cachedImages object: The key will always be a number, the value will always be an HTMLImageElement
```
Comment (tcasewiley) [inline src/my-element.ts:119]: OK so there was an issue when loading the images directly from the url: Every draw would make a new request to load the image. Obviously not good and was actually messing with the positioning of the elements.

We can use this cachedImages object to store all the images that will need to be rendered for this graph. Obviously should not be used for huge numbers of images, but for 2 or 10 or 50 should still be OK.

There are loadImage and getImage methods for storing and accessing images respectively.

File: src/my-element.ts [hunk 2]
```diff
@@ -175,6 +189,32 @@ export class MyElement extends LitElement {
   })
   quadrants = ['D', 'i', 'S', 'C']
 
+  constructor() {
```
Comment (tcasewiley) [inline src/my-element.ts:192]: This is being used as a stand-in for loading in mock data. The constructor will probably be removed once there is a proper pipeline for loading in users.

Comment (tcasewiley) [reply src/my-element.ts:192]: *definitely be removed

File: src/my-element.ts [hunk 3]
```diff
@@ -214,6 +254,37 @@ export class MyElement extends LitElement {
...
+    // const scale = this.height || this.width //scale
+
+    const radians = (angle) * (Math.PI / 180) //-90 orients upwards? So this is radians from the positive y axis?
+
+    const canvas = this._canvas.value as HTMLCanvasElement
+    const ctx = canvas.getContext('2d') as CanvasRenderingContext2D
+    ctx.save()
+    ctx.setTransform(1,0,0,1,0,0) // Something is messing with the transform outside this function, we're resetting it to default before beginning
+    ctx.translate(canvas.width/2, canvas.height/2) //set transform matrix to center of canvas
+    const offset = (this.mapRadius/2) * vector // This gives us the distance from center for our profile image
```
Comment (tcasewiley) [inline src/my-element.ts:260]: So the rotation from center ran into an issue where the images were being rotated as well. Obvi our UI team will be pretty upset with us if all the C and S style folks are looking at their profile pictures upside down.

We're using the same concept to determine placement as the SVG DiSC graph. Determine the offset from the center, use the offset as the radius of a circle, then use Math.sin and Math.cos to determine x and y coords on that circle, draw the profile avatar in that spot.

File: src/my-element.ts [hunk 4]
```diff
@@ -471,7 +543,8 @@ export class MyElement extends LitElement {
     graph.drawPriorityLabel('OBJECTIVITY', 225, 'center', textInside, false, kerning)
     graph.drawPriorityLabel('RELIABILITY', 180, 'center', textInside, false, kerning)
 
-
+    this._drawProfileAvatar(180, 2, 1)
+    this._drawProfileAvatar(125, 1.5, 2)
```
Comment (tcasewiley) [inline src/my-element.ts:534]: When drawing, we provide the angle, the vector, and the accountID of the user whose profile we want to draw. The accountID is used to load the correct image.

**Review by GirasolXimena — APPROVED** [review]

**Pull request merged**